test/new-e2e/tests/process/testing.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package process contains end-to-end tests for the general functionality of the process agent.
+package process
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	agentmodel "github.com/DataDog/agent-payload/v5/process"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+)
+
+//go:embed config/process_check.yaml
+var processCheckConfigStr string
+
+//go:embed config/process_discovery_check.yaml
+var processDiscoveryCheckConfigStr string
+
+//go:embed config/system_probe.yaml
+var systemProbeConfigStr string
+
+// assertRunningChecks asserts that the given process agent checks are running on the given VM
+func assertRunningChecks(t *assert.CollectT, vm client.VM, checks []string, withSystemProbe bool) {
+	status := vm.Execute("sudo datadog-agent status --json")
+	var statusMap map[string]any
```

Comment:
Maybe create an anonymous struct? https://dev.to/wagslane/use-anonymous-structs-for-json-marshalling-in-go-282g

Reply:
Oh nice, TIL!